### ISSUE TITLE
Retry m3u8 download on live HLS stream with no segments

### ISSUE THREAD

**Issue by GoogleCodeExporter** [oralodabas/google-cast-sdk#268]
```
This is more of an enhancement that a bug, since I know in this case that my 
playlist is (temporarily) invalid – but every other HLS player I've tried 
seems to accommodate this scenario.

What steps will reproduce the problem?
1. Start a live HLS stream with no segments available on load.
2. Load that URL via the Media Player Library
3. Player aborts.

What is the expected output? What do you see instead?

My *hope* would be that the Media Player library might retry the playlist 
download two or three times (with some delay between retries). This way, if it 
takes a few seconds for the first segment to become available, the video will 
still eventually load.  What I see instead is that the player seems to 
basically say "Well, this playlist is empty – so I  give up".

What version of the product are you using? On what operating system?

Media Player Library 0.6.0

```

Original issue reported on code.google.com by `kenni...@gmail.com` on 15 May 2014 at 2:09
